cloudinit/distros/debian.py

```diff
@@ -181,121 +152,12 @@ def _get_localhost_ip(self):
-            except subp.ProcessExecutionError:
-                # Even though we have already waited for the apt lock to be
-                # available, it is possible that the lock was acquired by
-                # another process since the check. Since apt doesn't provide
-                # a meaningful error code to check and checking the error
-                # text is fragile and subject to internationalization, we
-                # can instead check the apt lock again. If the apt lock is
-                # still available, given the length of an average apt
-                # transaction, it is extremely unlikely that another process
-                # raced us when we tried to acquire it, so raise the apt
-                # error received. If the lock is unavailable, just keep waiting
-                if self._apt_lock_available():
-                    raise
-                LOG.debug("Another process holds apt lock. Waiting...")
-                time.sleep(1)
-        raise TimeoutError("Could not get apt lock")
-
     def package_command(self, command, args=None, pkgs=None):
-        """Run the given package command.
-
-        On Debian, this will run apt-get (unless APT_GET_COMMAND is set).
-
-        command: The command to run, like "upgrade" or "install"
-        args: Arguments passed to apt itself in addition to
-              any specified in APT_GET_COMMAND
-        pkgs: Apt packages that the command will apply to
-        """
-        if pkgs is None:
-            pkgs = []
-
-        e = os.environ.copy()
-        # See: http://manpages.ubuntu.com/manpages/bionic/man7/debconf.7.html
-        e["DEBIAN_FRONTEND"] = "noninteractive"
-
-        wcfg = self.get_option("apt_get_wrapper", APT_GET_WRAPPER)
-        cmd = _get_wrapper_prefix(
-            wcfg.get("command", APT_GET_WRAPPER["command"]),
-            wcfg.get("enabled", APT_GET_WRAPPER["enabled"]),
-        )
-
-        cmd.extend(list(self.get_option("apt_get_command", APT_GET_COMMAND)))
-
-        if args and isinstance(args, str):
-            cmd.append(args)
-        elif args and isinstance(args, list):
-            cmd.extend(args)
-
-        subcmd = command
-        if command == "upgrade":
-            subcmd = self.get_option(
-                "apt_get_upgrade_subcommand", "dist-upgrade"
-            )
-
-        cmd.append(subcmd)
-
-        pkglist = util.expand_package_list("%s=%s", pkgs)
-        cmd.extend(pkglist)
-
-        self._wait_for_apt_command(
-            short_cmd=command,
-            subp_kwargs={"args": cmd, "env": e, "capture": False},
-        )
-
-    def update_package_sources(self):
-        self._runner.run(
-            "update-sources",
-            self.package_command,
-            ["update"],
-            freq=PER_INSTANCE,
-        )
+        # As of this writing, the only use of `package_command` outside of
+        # distros calling it within their own classes is calling "upgrade"
+        if command != "upgrade":
```

Comment:
-missing dist-upgrade too?  ... as well as any config overrides in apt_get_upgrade_subcommand looks like

Reply:
isn't dist-upgrade called full-upgrade these days?

Reply:
> -missing dist-upgrade too?

Not as far as I can see. There's no calling code that calls `package_command()` with anything other than `upgrade`.